middleware/auth.js

```diff
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken')
+const config = require('config')
+
+module.exports = function(req, res, next) {
+  // Get token from header
+  const token = req.header('x-auth-token')
+
+  // Check if not token
+  if(!token) {
+    return res.status(401).json({msg: 'No token, authorization denied'})
+  }
+
+  try {
+    const decoded = jwt.verify(token, config.get('jwtSecret'))
+
+    req.user = decoded.user
+    next()
+  } catch (error) {
+    res.status(401).json({msg: 'Token is not valid'})
```

Comment:
401 unauthorized because the user needs the correct user id and password in order to access the account.

Reply:
but in this case, we're catching all errors, even ones where it's not because they're not unauthorized.

For example, if the error was that `jwt.verify()` fails, we respond with a 401, which makes sense. But let's say our code fails because `config.get()` in line 14 throws an error. in that case, the user's credentials might have been correct. It's just that our config setup was wrong. In that scenario, we probably shouldn't throw a 401, and instead throw a 500.

401 says: your request was processed, but you're not allowed.
500 says: your request was not able to be processed.

I'd maybe replace the catch then with
```
catch(error) { 
  if (error instanceof <the error we care about>) {
    res.status(401).json({ msg: 'Token is not valid' })
  } else {
    res.status(500)
  }
}
```

Reply:
Ok